routes/items.js

```diff
@@ -0,0 +1,23 @@
+const express = require("express");
+const {
+  addTagsToItem,
+  removeTagsFromItem,
+  getItemsBasedOnType,
+  getItemsBasedOnTagId,
+  getAllItems,
+  getItemsBasedOnitemId,
+  getItemsBasedOnTagType,
+} = require("../controllers/items");
+const { validateItemsPayload } = require("../middlewares/validators/items");
+
+const router = express.Router();
+
+router.post("/", validateItemsPayload, addTagsToItem);
+router.delete("/", removeTagsFromItem);
+router.get("/bytype/:type", getItemsBasedOnType);
+router.get("/bytag/:tagid", getItemsBasedOnTagId);
+router.get("/byitem/:itemid", getItemsBasedOnitemId);
+router.get("/bytagtype/:tagtype", getItemsBasedOnTagType);
+router.get("/", getAllItems);
```

Comment:
I think all these endpoints can be removed and we can add these queries in the query params